### ISSUE TITLE
confusing about the precision calculation in many_to_many method Deteval.py 

### ISSUE THREAD

**Issue by fdengmark** [cs-chan/Total-Text-Dataset#14]
Hi I'm reading the Deteval.py script and I'm confusing about the precision calculation in the many_to_many() method. (line 203)

![image](https://user-images.githubusercontent.com/39723565/54811406-942fb900-4cc3-11e9-951e-07d55ef4d41f.png)

when you calculating recall, you considered the num_qualified_sigma_candidates, but you don't consider num_qualified_tau_candidates when you calculate precision. Moreover, given the following condition (line 199), I think this method really should be called as many_to_one instead of many_to_many. 

![image](https://user-images.githubusercontent.com/39723565/54811860-abbb7180-4cc4-11e9-82f9-02a0d96597d4.png)

In summary, I think if you don't consider num_qualified_tau_candidates when you calculate precision in many_to_many method, and you only check if np.sum(local_tau_table[qualified_sigma_candidates, det_id]) >= tp. This method really should be called as many_to_one, and you probably need another many_to_many method.

**Comment by fdengmark** [comment]
Regardless of this issue, I really appreciate your work and I learned a lot from your code. Thank you.

**Comment by ckchng** [comment]
Hi there, we noticed this issue 2 weeks ago and patched it in our latest commit. Thanks for pointing it out! 

Yes we did not consider the many-to-many case in the script, as the implementation is more complicated than the other cases and it is relatively rare in compared to other cases. However, we do plan to implement it in the near future. We would appreciate if you'd like to contribute.

Thank you for your kind words. =)

**Comment by fdengmark** [comment]
Sorry for the late reply, I’m glad if I can contribute to this project. I’m actually very new to github, so I don’t really know what is the contribution process. I will finish the many to many part before next week. Moreover, I have coded the precision recall computation as an method so you can call it with other script. (at least I think it is more convenient to use)
